mostlyai/engine/_tabular/generation.py

```diff
@@ -1014,9 +1186,25 @@ def generate(
 
                     # get seed data for current step
                     seed_step = seed_batch_grouped.nth(seq_step) if seq_step < n_seed_steps else pd.DataFrame()
-                    seed_step_encoded = (
-                        seed_batch_encoded_grouped.nth(seq_step) if seq_step < n_seed_steps else pd.DataFrame()
-                    )
+
+                    # drop NULL imputation columns from seed_step to allow conditional generation
+                    if imputation and len(seed_step) > 0:
+                        seed_step = _drop_fully_null_imputed_columns(seed_step, imputation.columns)
+
+                    # encode seed_step (after dropping NULL imputation columns)
+                    if len(seed_step) > 0:
+                        seed_step_encoded, _, _ = encode_df(
+                            df=seed_step, stats=tgt_stats, tgt_context_key=tgt_context_key
+                        )
+                    else:
+                        seed_step_encoded = pd.DataFrame()
+                    # filter seed_step to match step_ctx_keys (in case sequences ended in previous step)
+                    if len(seed_step) > 0:
+                        seed_step = seed_step[seed_step[tgt_context_key].isin(step_ctx_keys)].reset_index(drop=True)
+                    if len(seed_step_encoded) > 0:
+                        seed_step_encoded = seed_step_encoded[
+                            seed_step_encoded[seed_context_key_encoded].isin(step_ctx_keys)
+                        ].reset_index(drop=True)
```

Comment:
Is this logic needed with introduction of this feature? I don't fully see the need, but will check again